### ISSUE TITLE
[Bug]: panic when run "delete from t where a > 1;"

### ISSUE THREAD

**Issue by jiangxinmeng1** [matrixorigin/matrixone#9552]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93):
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

https://github.com/matrixorigin/matrixone/actions/runs/5020424215
<img width="567" alt="企业微信截图_16844783945825" src="https://github.com/matrixorigin/matrixone/assets/51114574/a0c96524-2b0e-482c-ab3e-17e8cc5e05be">

<img width="383" alt="e28d87ccf1846352dd81d433e97d79e" src="https://github.com/matrixorigin/matrixone/assets/51114574/e6ea20fd-c1e4-4b4f-932b-cf28865f29b0">

**Comment by JackTan25** [comment]
fix at #9556

**Comment by JackTan25** [comment]
@jiangxinmeng1 pls confirm it
